velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -316,6 +316,44 @@ struct ArraySumFunction {
   }
 };
 
+template <typename T>
+struct ArrayAverageFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      double& out,
+      const arg_type<Array<double>>& array) {
+    double sum = 0;
+    size_t count = 0;
+    for (const auto& item : array) {
+      if (item.has_value()) {
+        sum += *item;
+        count++;
+      }
+    }
+    if (count != 0) {
+      out = sum / count;
+    }
+    return count != 0;
+  }
+
+  FOLLY_ALWAYS_INLINE bool callNullFree(
+      double& out,
+      const arg_type<Array<double>>& array) {
+    // If the array is empty, then set result to null.
```

Comment:
Why do we not add a similar optimization : 
if (array.size() == 0) {
      return false;
    }
to the call function above also ?

Reply:
This is a good point. I added the check in the call function above also. Thanks!